x-pack/elastic-agent/pkg/artifact/config.go

```diff
@@ -47,7 +47,7 @@ func DefaultConfig() *Config {
 	return &Config{
 		SourceURI:       "https://artifacts.elastic.co/downloads/",
 		TargetDirectory: filepath.Join(homePath, "downloads"),
-		Timeout:         30 * time.Second,
+		Timeout:         60 * time.Second, // binaries are a bit larger it might take >30s to download them
```

Comment:
You used 120 above and 60 here, why the difference?

Reply:
this one is used for beats and endpoint, these binaries a smaller so it enables us to fail faster in case somethigns wrong. 
i'm considering 2 approaches: have agent timeout *2 of what this regular is set to
or set this to higher number but slow down processing loop in case server responds very slowly

Reply:
Will it really be slower? A TCP reset will not rely on timeout, it's only in the case the server stops pushing data in the HTTP response. I see that as a very rare-case.

Reply:
not really slower, but it takes more time to download 100megs than 40, we use this as a timecap for execution of the whole request. not only getting a response